minizip-1.1/zip.c

```diff
@@ -977,11 +980,25 @@ int Write_LocalFileHeader(zip64_internal* zi, const char* filename, uInt size_ex
 	  err = zip64local_putValue(&zi->z_filefunc,zi->filestream, LOCALHEADERMAGIC1, 4);
 	  if (err==ZIP_OK)
 	  {
-	  	err = zip64local_putValue(&zi->z_filefunc,zi->filestream, LOCALHEADERMAGIC2, 4);
+	  	err = zip64local_putValue(&zi->z_filefunc,zi->filestream, LOCALHEADERMAGIC2, 2);
 	  }
 	  if (err==ZIP_OK)
 	  {
-	  	err = zip64local_putValue(&zi->z_filefunc,zi->filestream, LOCALHEADERMAGIC3, 2);
+	  	char buf[5] = {0};
+	  	snprintf(buf, 5, "%02X%02X", tiapimajor, tiapiminor);
+	  	err = zip64local_putValue(&zi->z_filefunc,zi->filestream, buf[0], 1);
```

Comment:
```
ZWRITE64(zi->z_filefun, zi->filestream, buf, 4);
```

Reply:
Oh, I had tried the 4 bytes at the time and didn't work, but I guess with `ZWRITE64` it might... let me see.

Reply:
seems to work fine